sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -276,6 +279,16 @@ public String getHostname() {
         }
     }
 
+    @Override
+    public Mono<Map<Symbol, Object>> getRemoteProperties() {
+        this.remotePropertiesReference.compareAndSet(null, RetryUtil.withRetry(
```

Comment:
We don't need an atomic reference for this? The getEndpointStates() is a replayable Flux, and replays the last cached value... by then, the sender.getRemoteProperties() will be populated. Returning as-is is fine.

Reply:
Removed AtomicReference